### PR TITLE
Validating wiki form fields

### PR DESCRIPTION
**Pull request by lfilho**

### CODE REVIEW

File: lib/data-api-routes.js [hunk 1]
```diff
@@ -15,8 +16,13 @@ function addRoutes (router, sessionStore, db, docIndex) {
         try {
           data = JSON.parse(raw);
         }
-        catch (e) {
-          return sendError(e, 'invalid json');
+        catch (err) {
+          return sendError(res, err, 'invalid json');
```
Comment (lfilho) [inline lib/data-api-routes.js]: Fixed the method arguments and normalized the error var naming according to the other places in the file...

File: tests/lib/data-api-routes.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+'use strict';
```
Comment (lfilho) [inline tests/lib/data-api-routes.js]: @joshwnj what do you think of this testing approach?
It's not working right now, though: I haven't figured out a way of mocking oauth stuff. I have even changed `/index.js` to export the server + router but I wasn't able to get anywhere. So I reverted those experiments and left just the basic file, as you see here now. How should we go about this? Thanks

Comment (joshwnj) [inline tests/lib/data-api-routes.js]: I think we can do most of the testing we need to with unit tests.  Eg. we can test *just* the function that validates data, without having to do an http request. We already know http requests work so we don't need to test that part.  

Does that make it easier / clearer how to proceed?

Comment (lfilho) [inline tests/lib/data-api-routes.js]: Yes. I'm still adjusting myself to your much more frugal approach on how to code stuff :) :) :+1:

File: package.json [hunk 3]
```diff
@@ -41,7 +41,7 @@
     "build-css": "scss frontend/styles/main.scss > static/dist/main.css",
     "build-js": "browserify -t babelify frontend/main.js > static/dist/main.js",
     "build-watch": "npm run pre-build && npm run build-css && watchify -t babelify -d -v -o static/dist/main.js frontend/main.js",
-    "test": "tape tests/"
+    "test": "tape tests/**/*.js"
```
Comment (lfilho) [inline package.json]: Should we suffix our test files with `-test.js` or something?

**Comment by lfilho** [comment]
"[WIP]" removed. Unit tests done in favor of API testing

File: package.json (hunk 3 above)
Comment (joshwnj) [inline package.json]: For now we can do without it. We'll only need that if we have non-test files in the `tests/` directory, right?

**Pull request merged**